### ISSUE TITLE
GraphQl - Add to cart bundle product doesn't add all selected checkbox values

### ISSUE THREAD

**Issue by heerdt** [magento/magento2#32590]
Hello,

I am facing an issue when adding bundle products to the cart with GraphQl.
Basically, if the bundle product has a checkbox option with multiple items, when adding 2 or more items of this option to the cart, only 1 item is added. 

Can't tell since which version this start happening, but I tested on 2.4.1 and 2.4.2.

### Preconditions (*)
Magento 2.4.1 & 2.4.2
GraphQl API
Bundle products with a checkbox option with multiple items

### Steps to reproduce (*)
**All steps should use GraphQL**
- Create a new empty cart 
- Fetch the product details and retrieve the bundle options 
- Add product + required options + same option multiple checkbox items  to the cart using 'addProductsToCart' mutation 

### Expected result (*)
Product is added to the cart and all options selected are in the cart item.

### Actual result (*)
Product is added to the cart, the options are added but only one of the checkbox items is added to the cart item.

Queries:
**Create empty cart**
`mutation { createEmptyCart }`

**fetch product options**
```
query {
  products(filter: {sku: { eq: "24-WG080"}} ) {
    items{
      uid
      sku
      ... on BundleProduct {
          options: items {
              option_id
              title
              required
              type
              position
              sku
              value: options {
                  id
                  uid
                  quantity
                  position
                  is_default
                  price
                  price_type
                  can_change_quantity
                  title: label
              }
          }
      }
    }
  }
}
```
![image](https://user-images.githubusercontent.com/819013/112370984-f61dc180-8cd5-11eb-8fa2-bcd0344e9578.png)

**add to cart and retrieve added item**
```
mutation {
  addProductsToCart(
    cartId: "NSe5Dqzni1hAIUuobt9CXU290yTXATdT",
    cartItems: [
        {
            quantity: 1,
            sku: "24-WG080",
            selected_options: [
                "YnVuZGxlLzEvMS8x",     #radio option
                "YnVuZGxlLzIvNC8x",     #radio option
                "YnVuZGxlLzMvNS8x",     #radio option
                "YnVuZGxlLzQvOC8x",     #checkbox Sprite Foam Roller option
                "YnVuZGxlLzQvOS8x",     #checkbox Sprite Foam Roller option
                "YnVuZGxlLzQvMTAvMQ==", #checkbox Sprite Foam Roller option
                "YnVuZGxlLzQvMTEvMQ==", #checkbox Sprite Foam Roller option
                "YnVuZGxlLzQvMTIvMQ=="  #checkbox Sprite Foam Roller option
            ]
        }
    ]

  ) {
    cart {
      items {
        id
        product {
          name
          sku
        }
        quantity
        
            ... on BundleCartItem {
                bundle_options {
                    id
                    label
                    type
                    values {
                        id
                        label
                        price
                        quantity
                    }
                }
            }
      }
    }
  }
}
```

### Cart Result
![image](https://user-images.githubusercontent.com/819013/112371276-544aa480-8cd6-11eb-8c70-0dd6d8a76893.png)

Checkbox option in the cart has only 1 value, but it should have 5 as defined in the add to cart query

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.4/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by SilinMykola** [comment]
@magento I am working on this

**Comment by heerdt** [comment]
Hello @SilinMykola, Magento2 EE support already worked on this problem and sent me a solution. Attached the patch to fix this issue.
[MDVA-36713-EE_2.4.2.zip](https://github.com/magento/magento2/files/6663106/MDVA-36713-EE_2.4.2.zip)

**Comment by SilinMykola** [comment]
@heerdt Hi! Should I create PR from this patches or you you will do it yourself?

**Comment by heerdt** [comment]
hey @SilinMykola leave it to me, will send a MR later today! Thanks!

**Comment by heerdt** [comment]
@magento I am working on this

**Comment by heerdt** [comment]
@magento I am working on this

**Comment by engcom-Alfa** [comment]
Needs to be re-tested once in 2.4-develop, as the patch has been given as per above comments.

**Comment by engcom-Hotel** [comment]
@magento give me 2.4-develop instance

**Comment by engcom-Hotel** [comment]
@magento give me 2.4-develop instance

**Comment by engcom-Hotel** [comment]
@magento give me 2.4-develop instance

**Comment by engcom-Hotel** [comment]
Hello @heerdt,

We have tried to reproduce this issue in Magento 2.4-develop branch. But it is not reproducible for us, may be the issue has been already resolved. We request you to please try to reproduce it in the 2.4-develop branch and let us know if it is still reproducible for you.

We have run the below mutation to add Bundle product in the cart:
```
mutation {
  addProductsToCart(
    cartId: "YXE5atTyjacW06XCRxB4DG28uxQ4MRl0"
    cartItems: [
      {
        quantity: 1
        sku: "Bundle32590"
        selected_options: [
          "YnVuZGxlLzgvMTUvMQ==",
          "YnVuZGxlLzgvMTYvMQ==",
          "YnVuZGxlLzcvMTMvMQ=="
        ]
      }
    ]
  ) {
    cart {
      items {
        uid
        product {
          name
          sku
        }
        quantity
        ... on BundleCartItem {
          bundle_options {
            uid
            label
            type
            values {
              id
              label
              price
              quantity
            }
          }
        }
      }
    }
  }
}
```

And the response for the same is as follows with all items added (with checkbox items as well):
```
{
  "data": {
    "addProductsToCart": {
      "cart": {
        "items": [
          {
            "uid": "MTk=",
            "product": {
              "name": "Bundle32590",
              "sku": "Bundle32590"
            },
            "quantity": 1,
            "bundle_options": [
              {
                "uid": "YnVuZGxlLzc=",
                "label": "Option 1",
                "type": "radio",
                "values": [
                  {
                    "id": 13,
                    "label": "Test Product",
                    "price": 100,
                    "quantity": 1
                  }
                ]
              },
              {
                "uid": "YnVuZGxlLzg=",
                "label": "Option 2",
                "type": "checkbox",
                "values": [
                  {
                    "id": 15,
                    "label": "Wayfarer Messenger Bag",
                    "price": 45,
                    "quantity": 1
                  },
                  {
                    "id": 16,
                    "label": "Rival Field Messenger",
                    "price": 45,
                    "quantity": 1
                  }
                ]
              }
            ]
          }
        ]
      }
    }
  }
}
```

Below is the screenshot of the bundle product from the Admin Panel:
![image](https://user-images.githubusercontent.com/51681618/146359968-9036b9df-1f32-411c-92d9-54fae529c78c.png)

Please let us know if we missed anything.

Thanks

**Comment by engcom-Hotel** [comment]
Dear @heerdt,

We have noticed that this issue has not been updated for a period of 14 Days. Hence we assume that this issue is fixed now, so we are closing it. Please raise a fresh ticket or reopen this ticket if you need more assistance on this.

Regards

**Comment by cpartica** [comment]
re-opening this as it's important to fix even if stalled issue

**Comment by cpartica** [comment]
this was fixed in 2.4.4 and closing it again
